src/Console/CommandRunner.php

```diff
@@ -156,6 +174,22 @@ public function run(array $argv, ConsoleIo $io = null)
         return Shell::CODE_ERROR;
     }
 
+    /**
+     * Application bootstrap wrapper.
+     *
+     * Calls `bootstrap()` and `events()` if application implements `EventApplicationInterface`.
+     *
+     * @return void
+     */
+    protected function bootstrap()
+    {
+        $this->app->bootstrap();
+        if ($this->app instanceof EventApplicationInterface) {
+            $eventManager = $this->app->events($this->getEventManager());
```

Comment:
Could this result in event listeners being dropped if the application's bootstrap() attaches events to the application manager? Shouldn't this be passing in the application's event manager back into events() hook?

Reply:
CommandRunner's and Server's event managers are set to the app's instance in `setApp` https://github.com/cakephp/cakephp/pull/11692/files#diff-5746a5b6845a932f92018ff514ae2000R110

Reply:
I missed that. My bad.